src/Microsoft.ML.CpuMath/CpuMathUtils.netcoreapp.cs

```diff
@@ -761,5 +788,22 @@ public static void SdcaL1UpdateSparse(float primalUpdate, int count, ReadOnlySpa
                 }
             }
         }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static bool VectorizationMakesSense(int count)
+        {
+            if (Avx.IsSupported)
+            {
+                return count >= Vector256Alignment;
```

Comment:
I don't think using `Vector256Alignment` here makes sense. We should have a separate named constant for this.

I also believe that the characteristics between AVX and SSE code will differ quite a bit.
* `SSE` should never come with any perf or startup penalty
  * Using SSE code requires at least 4 elements (16-bytes of data)
  * The worst case scenario is that the user has an array and the first 4 elements cross a cache-line or page boundary; but that should be a fairly rare occurence
* `AVX` (256-bit vectorized code, to be more precise), however, can cause processor downclocking
  * The Intel Optimization manual has more details on the downclocking that can occur, and we should find the "sweet" spot for each workload

I would imagine that all the numbers you have above are from AVX code, and that commenting out the AVX code-path and testing explicitly for just VEX-encoded SSE would show better numbers.